main.rb

```diff
@@ -0,0 +1,98 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times { |i| yield(to_a[i]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    size.times do |i|
+      yield to_a[i], i
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    ary = []
+    my_each do |elem|
+      ary << elem if yield elem
+    end
+    ary
+  end
+
+  def my_all?
+    return nil unless block_given?
+
+    my_each do |elem|
+      return false unless yield elem
+    end
+    true
+  end
+
+  def my_map(proc = nil)
+    return dup unless block_given?
+    return dup if proc.nil?
+
+    ary = []
+    if proc.nil?
+      to_a.my_each { |elem| ary << yield(elem) }
+    else
+      to_a.my_each { |elem| ary << proc.call(elem) }
+    end
+    ary
+  end
+
+  def my_none?
+    return nil unless block_given?
+
+    my_each do |elem|
+      return false if yield elem
+    end
+    true
+  end
+
+  def my_any?
+    return nil unless block_given?
+
+    my_each do |elem|
+      return true if yield elem
+    end
+    false
+  end
+
+  def my_count
+    return size unless block_given?
+
+    n = 0
+    my_each do
+      n += 1
+      n
+    end
+  end
```

Comment:
- enumerables #my_count counts the number of items in enum that are equal to item if an argument is given   
- enumerables #my_count counts the number of elements yielding a true value if a block is given